### PR TITLE
Testing changes in preparation for no-input support

### PR DESCRIPTION
**Pull request by jtniehof**
EDIT: #26 includes everything in this PR...better to merge that one if there is time to review the whole business; otherwise I'll rebase #26 after this is merged.

This PR includes some rearrangement and testing support that I put in to prepare for the support of having processes with no input products (#9). There are some tests for basic database functionality when such a process exists (and has artificially-inserted output files) and the start of tests for DBRunner, before I start changing its functionality.

Because this involved some changing around of how the unit tests work, I figured it was better to get this in to master now, before I started in on actually changing DBRunner's functionality. I've been sort of working around this with other tests.

## PR Checklist

- [X] Pull request has descriptive title
- [X] Pull request gives overview of changes
- [X] New code has inline comments where necessary
- [X] (N/A) Any new modules, functions or classes have docstrings consistent with dbprocessing style
- [X] (N/A) Major new functionality has appropriate Sphinx documentation
- [X] (N/A) Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [X] New features and bug fixes should have unit tests
- [X] Relevant issues are linked in the description (e.g. `See issue #` or `Closes #`)

### CODE REVIEW

**Comment by jtniehof** [comment]
I'm going to wait on setting up CircleCI testing until this one is merged, since it makes the testing a little more robust.

**Review by dnadeau-lanl — APPROVED** [review]
Everthing looks fine.

**Pull request merged**
